include/unicore-mx/usbd/usbd.h

```diff
@@ -279,7 +284,7 @@ enum usbd_transfer_flags {
 	USBD_FLAG_PACKET_PER_FRAME_MASK = (0x3 << 5)
 };
 
-typedef enum usbd_transfer_flags usbd_transfer_flags;
```

Comment:
possibly i`ve got a warning here? enum and typedef - with same name. it`s confusing me, i newer see such trick. 
i`ve saw overriding enum item name with macro definition, but override enum by typedef - is it vlalid practice for C?